.github/workflows/validate_relaxng.yml

```diff
@@ -0,0 +1,22 @@
+name: validate relaxng
+
+on: [push]
+
+jobs:
+  validate-relaxng:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@main
+    - name: Count articles
+      run: echo '::set-output name=TOTAL::`ls articles/[0-9]*/*.xml | wc -l`'
+      id: article-count
+    - name: Output article count
+      run: echo "Validating ${{ steps.article-count.outputs.TOTAL }} articles" >> GITHUB_STEP_SUMMARY
+    - name: Enable xmllint matcher
+      uses: korelstar/xmllint-problem-matcher@v1
+    - name: Validate articles
+      uses: ChristophWurst/xmllint-action@master
+      with:
+        xml-file: articles/[0-9]*/*.xml
```

Comment:
Not sure how helpful the error messages are! I'm sure you would know better than I do. If time is not a factor (probably not) and there's another validator that has more useful output, I'm sure you could swap it.

Reply:
Ah, it just occurred to me you're not seeing the output of these workflows! I think you may have to enable something if you want them to run on your repo. Here are some links to these actions running on my fork so you can see what the output looks like. (They are old enough you can't see the steps, but you can see the output; LMK if you would like me to run them again on my fork so you can see what the steps look like.)

- [validate relax-ng](https://github.com/rlskoeser/dhq-journal/actions/runs/2373466097)
- [validate schematron](https://github.com/rlskoeser/dhq-journal/actions/runs/2373466031)

I see that you're right, it looks like the schematron validation is reporting problems with your schema and not with applying the schema.

Reply:
Hi @rlskoeser, it's nice to meet you, albeit virtually! Thanks for pointing us to the outputs, could you run them again in the fork so we can see the steps?

Reply:
couldn't figure out how to re-run from the interface, but amended my last commit message and force pushed to trigger, here are the new runs:

- [validate relax-ng](https://github.com/rlskoeser/dhq-journal/actions/runs/3411314115)
- [validate schematron](https://github.com/rlskoeser/dhq-journal/actions/runs/3411314157)

Reply:
Thank you @rlskoeser; good observation, @juliaflanders!

To check on my “`xmllint` reports yes|no just fine, but messages are near useless” theory I validated all .xml files in articles/[0-9]*/ against common/schema/DHQauthor-TEI.rng with each validator.[1] I then took the output of each and tweaked it for human readability. I then inserted extra vertical whitespace so that the files line up with each other if the two reports are read side-by-side. The results are really, really disappointing. Not only are the `xmllint` error messages often just plain bogus, in roughly a dozen cases it reports errors in a file `jing` considers valid.

Do you have the capability to simultaneously scroll 2 files side-by-side? If so, do so with the 2 files attached. (They are in [1 ZIP file](https://github.com/Digital-Humanities-Quarterly/dhq-journal/files/10009427/DHQ_validation_errors.zip).) If not, let me know and I will generate a single file that contains both sets of information. (It is not that hard, but it does require truncating the error messages in one file, and will take a little while, measured in a small number of minutes, not seconds.)

#### Notes
[1] Doing so with `xmllint` (now that @rlskoeser has shown me it can be done all-at-once) took a mere 4¼ s; doing so with `jing` (one at a time) took 3½ _minutes_.